python/pyarrow/tests/test_parquet.py

```diff
@@ -91,8 +87,14 @@ def test_pandas_parquet_2_0_rountrip(tmpdir):
 
     filename = tmpdir.join('pandas_rountrip.parquet')
     arrow_table = pa.Table.from_pandas(df, timestamps_to_ms=True)
+    expected_custom_metadata = {
+        b'indices': bytes([len(df.columns)]),
+        b'has_name': bytes([0])
+    }
```

Comment:
I think <= 255 levels is OK. I would actually rather see this metadata stored as a JSON blob under a single `pandas` key, otherwise we are possibly muddying the metadata namespace.

```
metadata = {b'pandas': json.dumps(pandas_meta).encode('utf8')}
```